### PR TITLE
Feature `bob git commit`

### PR DESCRIPTION
**Pull request by 06rajesh**

### CODE REVIEW

**Comment by Equanox** [comment]
I think this is not helpful.
image

What if we show the user in which repository a commit happened. If there is nothing to do, just print a similar message as plain `git commit` would do "nothing to commit, working tree clean"

File: bobgit/commit.go [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+	filteredRepo, untrackedRepo, err := filterModifiedRepos(repoNames)
+	errz.Fatal(err)
+
+	// throw some user message for untracked repositories
+	if len(filteredRepo) == 0 {
+		s := "nothing to commit, working trees are clean."
+		if len(untrackedRepo) > 0 {
+			s = "nothing added to commit but untracked files present."
+		}
+		boblog.Log.V(1).Info(s)
```
Comment (Equanox) [inline bobgit/commit.go:60]: Do not use boblog for user facing messages, should only be used if you want to log something. Using `fmt.Printf()` for cli output is totaly fine.

**Review by Equanox — CHANGES_REQUESTED** [review]

File: bobgit/commit.go [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+		if err != nil {
+			return usererror.Wrapm(err, "Failed to commit to git in repo "+name)
+		}
+	}
+
+	return nil
+}
+
+// filterModifiedRepos filters the repositories with changes
+// by running `git status` command on each repository and look for
```
[the remark this answers is not in the document]
Comment (06rajesh) [reply bobgit/commit.go]: changed

File: bobgit/commit.go (hunk 1 above)
Comment (06rajesh) [reply bobgit/commit.go]: removed

File: bobgit/commit.go [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+		if err != nil {
+			return updatedRepo, untrackedRepo, err
+		}
+
+		status, err := parse(output)
+		if err != nil {
+			return updatedRepo, untrackedRepo, err
+		}
+
+		tracked := false
```
[the remark this answers is not in the document]
Comment (06rajesh) [reply bobgit/commit.go]: changed

**Comment by 06rajesh** [comment]
> I think this is not helpful. image
> 
> What if we show in which repository a commit happened. If there is nothing to do, just print a similar message as plain `git commit` would do "nothing to commit, working tree clean"

returns a similar message as git. Also returns more information in case of untracked files.
```
s := "nothing to commit, working trees are clean."
		if len(untrackedRepo) > 0 {
			s = "nothing to commit, untracked files present in working tree."
}
```

**Review by Equanox — APPROVED** [review]

**Pull request merged**